src/objective-c/GRPCClient/private/NSData+GRPC.m

```diff
@@ -71,7 +71,7 @@ + (instancetype)grpc_dataWithByteBuffer:(grpc_byte_buffer *)buffer {
   }
   char *array;
   NSUInteger length;
-  MallocAndCopyByteBufferToCharArray(buffer, &length, &array);
+  MallocAndCopyByteBufferToCharArray(buffer, (size_t *)&length, &array);
```

Comment:
This fools `MallocAndCopyByteBufferToCharArray` into thinking it can write memory beyond what is allocated in the stack for `length` (`NSUInteger` is an `int` and `size_t` is a `long`). Because the size of the response is controlled by the server (and more so because `MallocAndCopyByteBufferToCharArray` is decompressing), it makes a server able to crash the client, which is a bug.